common/metrics/common.go

```diff
@@ -48,3 +54,22 @@ func getMetricDefs(serviceIdx ServiceIdx) map[int]metricDefinition {
 
 	return defs
 }
+
+// GetMetricsServiceIdx returns the metrics name
+func GetMetricsServiceIdx(serviceName string, logger log.Logger) ServiceIdx {
+	switch serviceName {
+	case primitives.FrontendService:
+		return Frontend
+	case primitives.HistoryService:
+		return History
+	case primitives.MatchingService:
+		return Matching
+	case primitives.WorkerService:
+		return Worker
+	default:
+		logger.Fatal("Unknown service name for metrics!", tag.Service(serviceName))
+	}
+
+	// this should never happen!
+	return NumServices
```

Comment:
why not just panic in the default?

Reply:
logger.Fatal already does panic. I'll add explicit panic as well.